File: paddler_client_python/paddler_client/conversation_message.py
from pydantic import BaseModel

from paddler_client.conversation_message_content_part import (
    ConversationMessageContentPart,
)

ConversationMessageContent = str | list[ConversationMessageContentPart]


class ConversationMessage(BaseModel):
    content: ConversationMessageContent
    role: str

File: paddler_client_javascript/src/schemas/ConversationMessage.ts
import { z } from "zod";

import { ConversationMessageContentPartSchema } from "./ConversationMessageContentPart";

export const ConversationMessageSchema = z.object({
  role: z.string(),
  content: z.union([
    z.string(),
    z.array(ConversationMessageContentPartSchema),
  ]),
});

export type ConversationMessage = z.infer<typeof ConversationMessageSchema>;
